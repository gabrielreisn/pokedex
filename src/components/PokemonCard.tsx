import React from "react";
import { PokemonCardProps } from "./types";
import { PokemonFrontCard } from "./PokemonFrontCard";
import { PokemonBackCard } from "./PokemonBackCard";

export const PokemonCard = ({
  number,
  name,
  types,
  attacks,
  maxCP,
  maxHP
}: PokemonCardProps) => {
  const [cardFace, toggleCard] = React.useState(true);

  const clickToggle = () => toggleCard(!cardFace);

  return cardFace ? (
    <PokemonFrontCard
      number={number}
      name={name}
      types={types}
      onClick={clickToggle}
    />
  ) : (
    <PokemonBackCard
      onClick={clickToggle}
      attacks={attacks}
      maxCP={maxCP}
      maxHP={maxHP}
    />
  );
};
