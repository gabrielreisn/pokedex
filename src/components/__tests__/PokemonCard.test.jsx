import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { PokemonCard } from "../PokemonCard";

const props = {
  attacks: {
    fast: [
      {
        damage: 12,
        name: "fast-attack-1",
        type: "Normal"
      }
    ],
    special: [
      {
        damage: 12,
        name: "special-attack-1",
        type: "Normal"
      }
    ]
  },
  maxCP: 123,
  maxHP: 123,
  name: "bulbasaur",
  number: "001",
  types: ["Gras,poison"]
};

describe("<PokemonCard />", () => {
  it("matches the snapshot", () => {
    const wrapper = render(<PokemonCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("initially should display the front card containning the pokemon image", () => {
    const { getByAltText } = render(<PokemonCard {...props} />);
    expect(getByAltText(`pokemon ${props.name}`)).toBeTruthy();
  });
  it("clicking on the pokemon should toggle the front and back cards", () => {
    const { getByAltText, container, getByText } = render(
      <PokemonCard {...props} />
    );
    const frontCard = getByAltText(`pokemon ${props.name}`);
    expect(frontCard).toBeTruthy();
    fireEvent.click(frontCard, container);
    const backCard = getByText("MAX CP");
    expect(backCard).toBeTruthy();
  });
});
