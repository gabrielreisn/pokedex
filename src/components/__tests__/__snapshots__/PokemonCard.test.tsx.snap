// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonCard /> matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" bg-gray-100 w-52 h-80 flex flex-col justify-around my-6 mx-auto rounded-lg border border-gray-500"
      >
        <img
          alt="pokemon bulbasaur"
          class="self-center w-full"
          loading="lazy"
          src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/1.png"
        />
        <div
          class="rounded-full py-0 px-3 bg-gray-400 w-16 text-center self-center mb-2"
        >
          <span
            class="relative font-semibold text-white"
          >
            #
            1
          </span>
        </div>
        <div
          class="bg-white flex items-start flex-col-reverse py-3 px-0 rounded-b-lg"
        >
          <div
            class="self-start my-1 mx-3"
          >
            <span
              class="rounded"
              style="background: rgb(120, 200, 80); padding: 1px 3px; margin: 0px 3px;"
            >
              Grass
            </span>
            <span
              class="rounded"
              style="background: rgb(104, 160, 144); padding: 1px 3px; margin: 0px 3px;"
            >
              poison
            </span>
          </div>
          <div
            class="self-start my-0 mx-3 font-light"
          >
            bulbasaur
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" bg-gray-100 w-52 h-80 flex flex-col justify-around my-6 mx-auto rounded-lg border border-gray-500"
    >
      <img
        alt="pokemon bulbasaur"
        class="self-center w-full"
        loading="lazy"
        src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/1.png"
      />
      <div
        class="rounded-full py-0 px-3 bg-gray-400 w-16 text-center self-center mb-2"
      >
        <span
          class="relative font-semibold text-white"
        >
          #
          1
        </span>
      </div>
      <div
        class="bg-white flex items-start flex-col-reverse py-3 px-0 rounded-b-lg"
      >
        <div
          class="self-start my-1 mx-3"
        >
          <span
            class="rounded"
            style="background: rgb(120, 200, 80); padding: 1px 3px; margin: 0px 3px;"
          >
            Grass
          </span>
          <span
            class="rounded"
            style="background: rgb(104, 160, 144); padding: 1px 3px; margin: 0px 3px;"
          >
            poison
          </span>
        </div>
        <div
          class="self-start my-0 mx-3 font-light"
        >
          bulbasaur
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
